dotcom-rendering/src/web/components/Byline.tsx

```diff
@@ -7,6 +7,7 @@ import { decidePalette } from '../lib/decidePalette';
 type Props = {
 	text: string;
 	format: ArticleFormat;
+	palette?: Palette;
```

Comment:
What's the purpose of passing around Palette here? Is this to do with sublinks styles?

Reply:
maybe passing around some kind of `format` and `containerFormat` props, then you can do something like `const palette = decidePalette(containerFormat || format)`

I don't know if I've looked at this PR in enough detail yet, but accepting that we maybe be reaching a point where a card may have to chose from multiple formats/palettes, getting the devx of this right is quite important!

Looking forward to seeing what we can come up with :)

Reply:
Well, in the end that change made a lot of sense, thanks for seeing it!

I've set things up so we're passing around `format` and `containerPalette` as the two 'types' which define the context of what is being rendered and then we expect the components to decide which palette to use. This is more declarative and seems to follow the existing patterns we've established.

However, it's a lot of changes being added to this PR but hopefully commit by commit it's manageable?

Reply:
Note. I'm very aware that `containerPalette` as a name doesn't work here. We can change it here in this PR but it might be easier as a follow on